### PR TITLE
Merge branch v3

### PR DESCRIPTION
**Pull request by Akshay-Kumar-S**

### CODE REVIEW

File: app/src/main/java/com/example/askapp/utils/Util.kt [hunk 1]
```diff
@@ -18,4 +20,17 @@ object Util {
...
+
+    fun isConnected(): Boolean {
+        val ctx: Context = AskSotiApplication.instance
+        val connectivityManager = ctx.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
+        return if (android.os.Build.VERSION.SDK_INT <= android.os.Build.VERSION_CODES.Q) {
+            var activeNetwork: NetworkInfo? = null
+                activeNetwork = connectivityManager.activeNetworkInfo
+                activeNetwork?.isConnectedOrConnecting == true
+        } else {
+            val capabilities = connectivityManager.activeNetwork
```
Comment (Akshay-Kumar-S) [inline app/src/main/java/com/example/askapp/utils/Util.kt:32]: optimize code

File: app/src/main/java/com/example/askapp/utils/Util.kt [hunk 2]
```diff
@@ -18,4 +20,17 @@ object Util {
             AskSotiApplication.instance, Manifest.permission.RECORD_AUDIO
         ) == PackageManager.PERMISSION_GRANTED
     }
+
+    fun isConnected(): Boolean {
+        val ctx: Context = AskSotiApplication.instance
+        val connectivityManager = ctx.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
+        return if (android.os.Build.VERSION.SDK_INT <= android.os.Build.VERSION_CODES.Q) {
+            var activeNetwork: NetworkInfo? = null
```
Comment (Akshay-Kumar-S) [inline app/src/main/java/com/example/askapp/utils/Util.kt:28]: optimize code

File: app/src/main/java/com/example/askapp/controllers/MainController.kt [hunk 3]
```diff
@@ -34,7 +34,7 @@ class MainController(private var mainInterface: IMainInterface) : IListener {
 
     override fun processUserQuery(query: String, queryType: Int) {
         GlobalScope.launch(Dispatchers.IO) {
-            if (CheckNetworkConnection().isConnect()) {
+            if (Util.isConnected()!!) {
```
Comment (Akshay-Kumar-S) [inline app/src/main/java/com/example/askapp/controllers/MainController.kt:37]: move to outside thread

Comment (KurienEapen) [reply app/src/main/java/com/example/askapp/controllers/MainController.kt:37]: Not possible

**Pull request merged**